addons/misra.py

```diff
@@ -1387,6 +1403,41 @@ def misra_7_1(self, rawTokens):
             if compiled.match(tok.str):
                 self.reportError(tok, 7, 1)
 
+    def misra_7_2(self, data):
+        # Large constant numbers that are assigned to a variable should have an
+        # u/U suffix if the variable type is unsigned.
+        def reportErrorIfMissingSuffix(variable, value):
+            if value and value.isNumber:
+                if variable and variable.valueType and variable.valueType.sign == 'unsigned':
+                    if variable.valueType.type in ['char', 'short', 'int', 'long', 'long long']:
+                        constantValue = parseUnsignedNumber(value.str, 0)
```

Comment:
Instead of this parsing.. I would prefer that you use `Valueflow::Value` instead.

In core Cppcheck the Token class has a few utility methods: `Token::hasKnownValue()` , `Token::hasKnownIntValue()` and `Token::getKnownIntValue()` .. I would like that the Token class in cppcheckdata.py also had such methods and then I would like that you used these instead here.

Reply:
Ok, will look into it.

Reply:
Thanks for the advice. I am now reading values using the value flow and I could remove my custom integer parser function. Should be available in my latest commit.